snuba/migrations/snuba_migrations/stacktraces/0001_stacktraces.py

```diff
@@ -0,0 +1,95 @@
+from typing import List, Sequence
+
+from snuba.clickhouse.columns import (
+    UUID,
+    Column,
+    DateTime,
+    String,
+    UInt,
+    NamedTuple,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+columns: List[Column[Modifiers]] = [
+    # primary key
+    Column("organization_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("transaction_id", UUID()),
+    Column("received", DateTime()),
+
+    # profiling data
+    Column("stacktrace", String(Modifiers(codecs=["LZ4HC(9)"]))),
+    Column("symbols", String(Modifiers(codecs=["LZ4HC(9)"]))),
+
+    # filtering data
+    Column("trace_id", UUID()),
+    Column("transaction_name", String(Modifiers(low_cardinality=True))),
+    Column("environment", String(Modifiers(nullable=True, low_cardinality=True))),
+
+    Column("version", NamedTuple((("name", String()), ("code", String())))),
+    Column("platform", String(Modifiers(low_cardinality=True))),
+    Column("android_api_level", UInt(32, Modifiers(nullable=True))),
+    Column("device_classification", String(Modifiers(low_cardinality=True))),
+    Column("device_locale", String(Modifiers(low_cardinality=True))),
+    Column("device_manufacturer", String(Modifiers(low_cardinality=True))),
+    Column("device_model", String(Modifiers(low_cardinality=True))),
+    Column("device_os_build_number", String(Modifiers(low_cardinality=True))),
+    Column("device_os_name", String(Modifiers(low_cardinality=True))),
+    Column("device_os_version", String(Modifiers(low_cardinality=True))),
+
+    Column("error_code", String(Modifiers(low_cardinality=True, nullable=True))),
+    Column("error_description", String(Modifiers(low_cardinality=True, nullable=True))),
+
+    Column("duration_ns", UInt(64)),
+
+    # internal data
+    Column("retention_days", UInt(16)),
+]
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+
+    def forwards_local(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=StorageSetKey.STACKTRACES,
+                table_name="stacktraces_local",
+                columns=columns,
+                engine=table_engines.MergeTree(
```

Comment:
IIRC you had this as replacing merge tree earlier and changed it to merge tree. Just curious about the reason behind that. It means we would not be able to do any updates or deletions nor deduplicate potentially duplicate inserts.

Reply:
I don't think we need the `ReplacingMergeTree` features as we ingest only 1 profile per transaction and everything is sorted with the transaction ID in the sorting key. But maybe it still be a good idea in case we ingest twice the same data (which wouldn't be needed, just a "bug" or a client wrongly sending twice the same transaction).

Reply:
I think ReplacingMergeTree might be safer even if are not going to support merging and deletions per se, for the reasons you mentioned. We generally do try to deduplicate events in ingest but some still make their way to ClickHouse. Also for compliance (gdpr?) we may want a way to remove data on customer request prior to the ttl. If the TTL is pretty short anyway it may not be a big deal as data will get automatically removed before long, but might be good to have the option to delete still.